File: v131016/TeeNew/Series_BarNegativeStacked.pas
unit Series_BarNegativeStacked;
{$I TeeDefs.inc}

interface

uses
  {$IFNDEF LINUX}
  Windows, Messages,
  {$ENDIF}
  SysUtils, Classes,
  {$IFDEF CLX}
  QGraphics, QControls, QForms, QDialogs, QExtCtrls, QStdCtrls, QComCtrls,
  {$ELSE}
  Graphics, Controls, Forms, Dialogs, ExtCtrls, StdCtrls, ComCtrls,
  {$ENDIF}
  Base, TeEngine, Series, TeeProcs, Chart, TeeTools, TeeComma;

type
  TBarStackNegative = class(TBaseForm)
    Series1: TBarSeries;
    Series2: TBarSeries;
    Series3: TBarSeries;
    TeeCommander1: TTeeCommander;
    ChartTool1: TColorLineTool;
    procedure FormCreate(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

implementation

{$IFNDEF CLX}
{$R *.DFM}
{$ELSE}
{$R *.xfm}
{$ENDIF}

procedure TBarStackNegative.FormCreate(Sender: TObject);
begin
  inherited;
  Series1.AddArray([ 23, -56, 42, 9, -8] );
  Series2.AddArray([ 32, -16, 9, 39, -28] );
  Series3.AddArray([ 12, -21, 14, 22, -13] );
end;

initialization
  RegisterClass(TBarStackNegative);
end.

File: TeeNew/Builder2009/Series_BarNegativeStacked.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#include <sysopen.h>
#pragma hdrstop

#include "Series_BarNegativeStacked.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base"
#pragma link "TeeComma"
#pragma link "TeeTools"
#pragma resource "*.dfm"
TBarStackNegative *BarStackNegative;
//---------------------------------------------------------------------------
__fastcall TBarStackNegative::TBarStackNegative(TComponent* Owner)
        : TBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TBarStackNegative::FormCreate(TObject *Sender)
{

  double vals1[5] = {23, -56, 42, 9, -8};
  Series1->AddArray(vals1,4);
  double vals2[5] = {32, -16, 9, 39, -28};
  Series2->AddArray(vals2,4);
  double vals3[5] = {12, -21, 14, 22, -13};
  Series3->AddArray(vals3,4);
}
//---------------------------------------------------------------------------

